dagger/dsl/node_invocation_recorder.py

```diff
@@ -179,6 +185,7 @@ def dsl(a):
                 preset_params[argument_name] = argument_value
 
         if preset_params:
+            # TODO @lorenzo can we review this? no test coverage and maybe has a bug
```

Comment:
@larribas after investigating why there are errors with default values and more complex scenarios I think I found something. This produces a function with signature `args` and `kwargs` that later fails on `_build`. Also I think this is currently not test covered. Maybe we can pair and review this together.